rust/kcl-lib/src/unparser.rs

```diff
@@ -794,30 +815,28 @@ impl Node<PipeExpression> {
             .map(|(index, statement)| {
                 let indentation = options.get_indentation(indentation_level + 1);
                 let mut s = statement.recast(options, indentation_level + 1, ExprContext::Pipe);
-                let non_code_meta = self.non_code_meta.clone();
+                let non_code_meta = &self.non_code_meta;
                 if let Some(non_code_meta_value) = non_code_meta.non_code_nodes.get(&index) {
```

Comment:
Removing this `clone` would be good, but seems unrelated to adding `Cow`. If part of your goal is to show that migrating to `Cow` improved performance by X%, then including changes like this goes against that.

So perhaps include this in a separate PR with similar cleanups?